### DISCUSSION TITLE
Vue playground for md-enhance

### DISCUSSION THREAD

**Discussion by FuckDoctors** [vuepress-theme-hope/vuepress-theme-hope#1923] [Q&A] [answered]
之前说的md增强playground插件的事，我按自己的想法写了一下。
[点击查看演示](https://vuepress-theme-hope-git-playground-fuckdoctors.vercel.app/md-enhance/guide/playground.html)

实现代码：<https://github.com/FuckDoctors/vuepress-theme-hope/tree/playground>

分为两种方式：
- 外置方式
   使用iframe嵌入外部playground，比如嵌入 Vue 官方的 playground，也可以嵌入第三方，也可以嵌入自己定制的playground
- 内置方式
   直接使用 `@vue/repl` 渲染出来。

两种方式均可个别自定义配置。

`@vue/repl`公开的东西太少，只能直接使用它的 `Repl.vue`，配置不是那么灵活。

@Mister-Hope 大佬看看行不行，看看有啥好的建议或意见。

**Comment by FuckDoctors** [comment]
基本使用：

````md
:::: playground Playground demo

::: file App.vue

```vue
<script setup>
import { ref } from "vue";
const msg = ref("Hello World!");
</script>
<template>
  <h1>{{ msg }}</h1>
  <input v-model="msg" />
</template>
```

:::
::::
````

内置模式：

````md
:::: playground Internal mode
::: file App.vue

```vue
<script setup>
import { ref } from "vue";
const msg = ref("Hello Playground!");
</script>
<template>
  <h1>{{ msg }}</h1>
  <input v-model="msg" />
</template>
```

:::
::: settings

```json
{
  "mode": "internal"
}
```

:::
::::
````

**Comment by Mister-Hope** [comment]
I think this is great, but I would like to see a more comfortable syntax like codetabs instead of nesting containers.

If you finished this feature,just call me and I will merge this in to another branch,band I will make some improvements later.

Also a more detailed docs and js docs on interface would be great.

**Comment by FuckDoctors** [comment]
@Mister-Hope Almost done.

I added playground docs and js docs with my poor english.
Docs preview: <https://vuepress-theme-hope-git-playground-fuckdoctors.vercel.app/md-enhance/>

Todo:
- [x] dark mode

Known issues:
- Code editor (CodeMirror) does not correctly calculate the height in internal mode.
   Once you clicked it, or changed the widow size, it will be correct.

**Comment by FuckDoctors** [comment]
By the way, I personally prefer to use nested containers, which is normal markdown-like😊.
It's a bit redundant indeed...

**Reply by Mister-Hope** [reply]
A nested container can be a bit confusing, as your file container means NOTHING when it's not inside playground container. So I insist we should avoid container syntax which it's not actually a standalone container.

Also `@xxx` is easier to read and write.

**Reply by FuckDoctors** [reply]
Yes, `@file` `@settings` `@imports` are good suggestions.

**Comment by Mister-Hope** [comment]
https://github.com/vuepress-theme-hope/vuepress-theme-hope/pull/1928

**Reply by FuckDoctors** [reply]
I will check it.

**Comment by Mister-Hope** [comment, marked as answer]
Rebuilt in beta99
